main.go

```diff
@@ -114,9 +116,11 @@ func (w *Worker) ConnectIfNeeded(ctx context.Context) error {
 	if w.conn == nil {
 		return w.ConnectWithRetries(ctx)
 	}
-	//If not connected to the desired target, try reconnecting if it's been 5 minutes
-	if !w.isConnectedToPrimary() && time.Since(w.lastReconnect) > 5*time.Minute {
-		log.Printf("Worker %d: attempting to reconnect to primary target", w.id)
+
+	// Reconnect if it's been 5 minutes
```

Comment:
Thanks yes @ymedentsii pointed out my failure to close before issuing the ConnectoWithRetries. Hopefully this fixes the memory leak, I'll test overnight.

Reply:
The original code only reconnected if it had failed over to another target. However in Kubernetes, if the target is the log-processor service, and log-processor has only 1 replica, it'll stay connected to that replica. If we scale log-processor to 10 replicas, the connections will not re-establish and all the traffic will still route to the same replica. The periodic reconnection is my attempt to use Kubernetes' service balancing to spread the worker connections across all the target replicas even if we scale up.

Reply:
Ah, I understand. Assuming the log-processor is manually scaled, and infrequently at that, I would suggest that the list of targets (of which I presume there is only 1, the log processor service) and connections per target (the number of log-processor replicas) be read from environment variables so that the connection fanout here can be matched to the log-processor replicas scale. Then I think this code would no longer be required as when the env var gets updated k8s will restart the pod.